File: src/model/pieces/PieceImplChess.java
package model.pieces;

import model.enums.ChessPieceType;

public class PieceImplChess extends Piece {

    ChessPieceType pieceType;

    public PieceImplChess() {
        super();
        pieceType = ChessPieceType.NONE;
    }

    public void setPieceType(ChessPieceType pieceType) {
        this.pieceType = pieceType;
    }

    public ChessPieceType getPieceType() {
        return pieceType;
    }

    @Override
    public void setPiece(Piece piece) {
        if(!(piece instanceof PieceImplChess pieceImplChess)) {
            throw new IllegalArgumentException();
        }
        this.pieceType = pieceImplChess.pieceType;
        this.setPlayer(pieceImplChess.getPlayer());
    }
    @Override
    public int getCode() {
        return 0;
    }
}

File: src/model/pieces/PieceImplChess.kt
package model.pieces

import model.enums.ChessPieceType

class PieceImplChess : Piece() {
    @JvmField
    var pieceType: ChessPieceType

    init {
        pieceType = ChessPieceType.NONE
    }

    public override fun setPiece(piece: Piece?) {
        require(piece is PieceImplChess)
        this.pieceType = piece.pieceType
        this.player = piece.player
    }

    override val code: Int
        get() = 0
}
